test/mocha/10-canonize-hash-api.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2017-2021 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const {util: {clone}} = require('bedrock');
+const brLedgerNode = require('bedrock-ledger-node');
+const mockData = require('./mock.data');
+
+const rdfCanonizeAndHash = brLedgerNode.consensus._rdfCanonizeAndHash;
+
+describe('Canonize And Hash API', function() {
+  describe('WebLedgerConfiguration', function() {
+    it('should canonize & hash', async function() {
+      const data = clone(mockData.ledgerConfiguration);
+      const result = await rdfCanonizeAndHash(data);
+      should.exist(result);
+      result.should.be.an('object');
+      result.should.have.property('canonizedBytes');
+      result.should.have.property('hash');
+      result.should.eql(mockData.canonize.ledgerConfiguration);
+    });
+    it('should have a unique hash', async function() {
```

Comment:
IMO, the utility of this test is questionable. A test that checks if a single property changed once is not sufficient to test that hashes from a function are unique .

Reply:
Could explain further here? A hash should definitely change if a single property is changed. What were testing here is that when the properties change the hash changes. I guess more accurately we're testing here that if a property's value changes, the hash changes, which means whatever functionality the hash is used for further along in the consensus process, it would pick up on the change.

Reply:
Yes, this test is not exhaustive and is a poor way to check for the uniqueness of a hash. Without diving too deep, it is not enough to check one sample input to ensure that unique hashes are guaranteed. One would need to enumerate all inputs and ensure there are no duplicate outputs. I'd argue that this kind of test is excessive. Hence, when testing one sample out of the set of many samples and asserting it's a test that ensures a unique hash... the utility of the test is questionable. I'm fine with this being a smoke test.

Reply:
It's definitely a smoke test, the point of it is to ensure that the positive test is not a false positive. It's not intended as an exhaustive check of a hashing function, but simply to ensure that different properties do create different hashes. That in turn gives us some information towards knowing if the positive test is a false positive.